src/server/cluster/incoming_slot_migration.cc

```diff
@@ -215,7 +207,8 @@ bool IncomingSlotMigration::Join(long attempt) {
 }
 
 void IncomingSlotMigration::Stop() {
-  string_view log_state = state_.load() == MigrationState::C_FINISHED ? "Finishing" : "Cancelling";
+  util::fb2::LockGuard lk(state_mu_);
+  string_view log_state = state_ == MigrationState::C_FINISHED ? "Finishing" : "Cancelling";
```

Comment:
also when the migration is stopped can we have some state that will show that the migration is stopped otherwise we will show the last state - sync/connecting